File: backend/src/main/java/com/oose/labsafety/education/domain/EducationCompletionResult.java
package com.oose.labsafety.education.domain;

import com.oose.labsafety.common.domain.Identifiable;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;

public record EducationCompletionResult(
        @NotBlank(message = "교육이수결과 ID는 필수입니다.") String completionResultId,
        @NotBlank(message = "연구활동종사자 ID는 필수입니다.") String researcherId,
        @NotBlank(message = "교육개설정보 ID는 필수입니다.") String openingId,
        String learningResultId,
        String logId,
        @NotBlank(message = "이수일은 필수입니다.") String completionDate,
        @Min(value = 1, message = "인정시간은 1 이상이어야 합니다.") int recognizedHours,
        @NotBlank(message = "이수 상태는 필수입니다.") String completionStatus,
        boolean manualRegistration
) implements Identifiable {
    @Override
    public String id() {
        return completionResultId;
    }
}

File: src/features/education/domain/EducationCompletionResult.js
import { createEntity } from '../../../shared/domain/createEntity';

const schema = {
  completionResultId: { label: '교육이수결과 ID', required: true },
  researcherId: { label: '연구활동종사자 ID', required: true },
  openingId: { label: '교육개설정보 ID', required: true },
  learningResultId: { label: '학습결과 ID' },
  logId: { label: '안전교육일지 ID' },
  completionDate: { label: '이수일', required: true },
  recognizedHours: { label: '인정시간', required: true, type: 'number', min: 1 },
  completionStatus: { label: '이수 상태', required: true },
  manualRegistration: { label: '수동 등록 여부', type: 'boolean' },
};

export function createEducationCompletionResult(input) {
  return createEntity(input, schema);
}
